### PR TITLE
Fix/ awk bug and docker CLI 🚧🚧🚧

### PR DESCRIPTION
**Pull request by mrnicegyu11**
This PR:
- explicitly installes `gawk` which then substitutes `mawk`. Our tooling expects `gawk`, which has different CLI options than `mawk`.
- Removes the `apt`-installed `docker` and adds the proper `docker-ce` and `docker-ce-cli`

🚧🚧🚧 :
Release this as `0.10.6` to fix a bug introduced by the usage of `docker run` in the Makefile in https://github.com/ITISFoundation/osparc-ops-environments/pull/99

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -68,6 +71,14 @@ FROM base as build
 
 ENV SC_BUILD_TARGET=build
 
+RUN curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
+  echo \
+  "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/debian \
+  $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null && \
+  apt-get update && \
+  apt-get install -y --no-install-recommends \
+  docker-ce docker-ce-cli containerd.io
```
Comment (GitHK) [inline Dockerfile:80]: You might want to fix the docker version here, or is it not important?

**Review by GitHK — APPROVED** [review]
OK for me.

**Review by pcrespov — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (sanderegg) [reply Dockerfile:80]: and by the way, maybe think of installing docker compose v2... not sure that comes automatically in.

**Review by sanderegg — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (mrnicegyu11) [reply Dockerfile:80]: @GitHK : Not important I think as the host's docker socket is used
@sanderegg : will adress

**Comment by mrnicegyu11** [comment]
--> released as `0.10.6`

**Pull request merged**